bouncer/tests/btc_utxo_consolidation.ts

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env -S pnpm tsx
+import assert from 'assert';
+
+import { submitGovernanceExtrinsic } from '../shared/cf_governance';
+import { provideLiquidity } from '../shared/provide_liquidity';
+import { getChainflipApi, runWithTimeout, sleep } from '../shared/utils';
+
+const chainflip = await getChainflipApi();
+
+async function queryUtxos(): Promise<{ amount: number; count: number }> {
+  const utxos: [{ amount: number }] = (
+    await chainflip.query.environment.bitcoinAvailableUtxos()
+  ).toJSON();
+
+  return {
+    amount: utxos.reduce((acc, utxo) => acc + utxo.amount, 0),
+    count: utxos.length,
+  };
+}
+
+async function test() {
+  const initialUtxos = await queryUtxos();
+
+  console.log(`Initial utxo count: ${initialUtxos.count}`);
+
+  if (initialUtxos.count === 0) {
+    throw new Error('Test precondition violated: btc vault should have at least 1 utxo');
+  }
+
+  // Reset consolidation parameters to ensure consolidation doesn't trigger immediately:
+  await submitGovernanceExtrinsic(
+    chainflip.tx.environment.updateConsolidationParameters({
+      consolidationSize: 100,
+      consolidationThreshold: 200,
+    }),
+  );
+
+  // Setting the threshold to current utxo count + 2 allows us to test the more
+  // general case of consolidationSize != consolidationThreshold even when there
+  // is only 1 UTXO available initially
+  const consolidationSize = initialUtxos.count + 1;
+  const consolidationThreshold = initialUtxos.count + 2;
+
+  await provideLiquidity('BTC', 2);
+  await provideLiquidity('BTC', 3);
```

Comment:
wasn't clear to me initially that these are related to the +2 above - a little comment would help